docs/fastdds/dds_layer/subscriber/dataReader/readingData.rst

```diff
@@ -0,0 +1,35 @@
+.. _dds_layer_subscriber_accessreceived:
+
+Accessing received data
+=======================
+
+The :ref:`dds_layer_subscriber_dataReader` informs through the
+:ref:`dds_layer_subscriber_dataReaderListener` when it receives new data values from any matching
+:ref:`dds_layer_publisher_dataWriter`, so that the application knows there is new data to process.
+The application can then access and consume the received data values.
+
+This access can be done by *reading* or *taking*.
+
+ * **Reading** is done with the :cpp:func:`eprosima::fastdds::dds::DataReader::read_next_data` member function.
+   It reads and returns the next, non-previously accessed data value available on the
```

Comment:
In general in the section, the return is a `RETURN_CODE`. It should be clarified somehow that the data and meta-data are not returned (strictly speaking), bu are populated in the function arguments (passed by ref) instead.